Rspecを用いたモデルのテスト

kategamo opened the pull request:
<WHAT> Rspecを用いたモデルのテスト
テスト内容
メッセージを保存できる場合
メッセージがあれば保存できる
画像があれば保存できる
メッセージと画像があれば保存できる
メッセージを保存できない場合
メッセージも画像も無いと保存できない
group_idが無いと保存できない
user_idが無いと保存できない
<WHY>「プログラムが意図した通りに動くことを確かめる」ため

kategamo committed:
Rspecを用いたモデルのテスト

メッセージを保存できる場合
メッセージがあれば保存できる
画像があれば保存できる
メッセージと画像があれば保存できる
メッセージを保存できない場合
メッセージも画像も無いと保存できない
group_idが無いと保存できない
user_idが無いと保存できない

kategamo commented:
<img width="1247" alt="screen shot 2017-12-16 at 9 31 47" src="https://user-images.githubusercontent.com/33300513/34065708-36a877b8-e248-11e7-91f6-9c22afa4e3e1.png">

yuya1212h4 commented on spec/controllers/messages_controller_spec.rb:
```diff
@@ -0,0 +1,90 @@
+
+    context 'not login'
+
+        it 'redirects to new_user_session_path' do
```
ここのインデントを合わせるようにお願いします 。

yuya1212h4 commented on spec/factories/groups.rb:
```diff
@@ -0,0 +1,7 @@
+  factory :group do
+    name Faker::Name.name
+    # created_at { Faker::Time.between(5.days.ago, 3.days.ago, :all) }
+    # updated_at { Faker::Time.between(2.days.ago, 1.days.ago, :all) }
```
必要がないのであれば、削除の方をよろしくお願い致します。

yuya1212h4 commented on spec/factories/messages.rb:
```diff
@@ -0,0 +1,12 @@
+FactoryGirl.define do
+  factory :message do
+    #body   { Faker::StarWars.wookie_sentence }
```
こちらのファイルについても、不必要な部分については削除をお願いします。

yuya1212h4 commented on spec/models/message_spec.rb:
```diff
@@ -0,0 +1,35 @@
+      end
+
+      it "is invalid without a group_id" do
+        message = build(:message,group_id: nil)
```
, の後にスペースを入れて頂くようにお願い致します。

kategamo committed:
モデルのテスト(2)

kategamo committed:
コントローラーのテスト

1.メッセージ一覧ページを表示するアクション
ログインしている場合
アクション内で定義しているインスタンス変数があるか
該当するビューが描画されているか
ログインしていない場合
意図したビューにリダイレクトできているか
2.メッセージを作成するアクション
ログインしているかつ、保存に成功した場合
メッセージの保存はできたのか
意図した画面に遷移しているか
ログインしているが、保存に失敗した場合
メッセージの保存は行われなかったか
意図したビューが描画されているか
ログインしていない場合
意図した画面にリダイレクトできているか

kategamo commented:
コントローラーのテスト
<img width="857" alt="screen shot 2017-12-18 at 12 42 33" src="https://user-images.githubusercontent.com/33300513/34089111-418b4110-e3f1-11e7-9d68-f891a4dec6b8.png">
